### DISCUSSION TITLE
Splitting a large object

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1303] [Q&A] [answered]
Let's say I have an atom containing initialData as in this [example](https://jotai.org/docs/advanced-recipes/large-objects).

How could I create from this atom  a read-only atom which contains only names?

[{ name: 'Luke Skywalker’},  { name: 'C-3PO' }]

**Comment by dai-shi** [comment, marked as answer]
This?
```js
const dataAtom = atom(initialData)
const nameAtom = atom((get) => get(dataAtom).people.map(({ name }) => ({ name })))
```
